app/navigation/ReturnsJourneyNavigator.scala

```diff
@@ -73,36 +77,36 @@ class ReturnsJourneyNavigator {
       case _ => throw new Exception("Unable to navigate to page")
     }
 
-
-  private def manufacturedPlasticPackagingRoute(answers: UserAnswers, mode: Mode = NormalMode): Call =
-    answers.get(ManufacturedPlasticPackagingPage) match {
-      case Some(true) => routes.ManufacturedPlasticPackagingWeightController.onPageLoad(mode)
-      case Some(false) =>
-        if (mode == CheckMode) {
-          routes.ConfirmPlasticPackagingTotalController.onPageLoad
-        } else {
-          routes.ImportedPlasticPackagingController.onPageLoad(mode)
-        }
-      case _ => throw new Exception("Unable to navigate to page")
-    }
+  def manufacturedPlasticPackagingRoute(mode: Mode, hasAnswerChanged: Boolean, usersAnswer: Boolean): Call = 
+    if (hasAnswerChanged) 
+      (mode, usersAnswer) match {
+        case (_, true)       => routes.ManufacturedPlasticPackagingWeightController.onPageLoad(mode)
+        case (CheckMode, _)  => routes.ConfirmPlasticPackagingTotalController.onPageLoad
+        case (NormalMode, _) => routes.ImportedPlasticPackagingController.onPageLoad(mode)
+    } else 
+      routes.ConfirmPlasticPackagingTotalController.onPageLoad
 
   private def manufacturedPlasticPackagingWeightRoute(answers: UserAnswers): Call =
     answers.get(ManufacturedPlasticPackagingWeightPage) match {
       case Some(_) => routes.ConfirmPlasticPackagingTotalController.onPageLoad
       case _ => throw new Exception("Unable to navigate to page")
     }
 
-  private def importedPlasticPackagingRoute(answers: UserAnswers, mode: Mode = NormalMode): Call =
-    answers.get(ImportedPlasticPackagingPage) match {
-      case Some(true) => routes.ImportedPlasticPackagingWeightController.onPageLoad(mode)
-      case Some(false) =>
-        if (mode == CheckMode) {
-          routes.ConfirmPlasticPackagingTotalController.onPageLoad
-        } else {
-          routes.ConfirmPlasticPackagingTotalController.onPageLoad
-        }
-      case _ => throw new Exception("Unable to navigate to page")
+  def importedPlasticPackagingRoute(answers: UserAnswers, mode: Mode = NormalMode, answerChanged: AnswerChanged = false): Call = {
+    if (mode == CheckMode) {
+      answers.get(ImportedPlasticPackagingPage) match {
+        case Some(true) if answerChanged => routes.ImportedPlasticPackagingWeightController.onPageLoad(mode)
+        case Some(false) | Some(true) if !answerChanged => routes.ConfirmPlasticPackagingTotalController.onPageLoad
+        case _ => throw new Exception("Unable to navigate to page")
+      }
+    } else {
+      answers.get(ImportedPlasticPackagingPage) match {
+        case Some(true) => routes.ImportedPlasticPackagingWeightController.onPageLoad(mode)
+        case Some(false) => routes.ConfirmPlasticPackagingTotalController.onPageLoad
+        case _ => throw new Exception("Unable to navigate to page")
+      }
     }
```

Comment:
Had a quick stab at shrinking this

  def importedPlasticPackagingRoute(answers: UserAnswers, mode: Mode = NormalMode, answerChanged: AnswerChanged = false): Call = {
    answers.get(ImportedPlasticPackagingPage) match {
      case Some(false) => routes.ConfirmPlasticPackagingTotalController.onPageLoad
      case Some(true) => (answerChanged, mode) match {
        case (false, CheckMode) => routes.ConfirmPlasticPackagingTotalController.onPageLoad
        case _ => routes.ImportedPlasticPackagingWeightController.onPageLoad(mode)
      }
      case _ => throw new Exception("Unable to navigate to page")
    }
  }